src/proto/media/file.rs

```diff
@@ -183,30 +372,48 @@ pub(super) struct FileImpl {
     ) -> Status,
     close: extern "win64" fn(this: &mut FileImpl) -> Status,
     delete: extern "win64" fn(this: &mut FileImpl) -> Status,
-    read:
-        extern "win64" fn(this: &mut FileImpl, buffer_size: &mut usize, buffer: *mut u8) -> Status,
-    write: extern "win64" fn(
+    read: unsafe extern "win64" fn(
+        this: &mut FileImpl,
+        buffer_size: &mut usize,
+        buffer: *mut u8,
+    ) -> Status,
+    write: unsafe extern "win64" fn(
         this: &mut FileImpl,
         buffer_size: &mut usize,
         buffer: *const u8,
     ) -> Status,
     get_position: extern "win64" fn(this: &mut FileImpl, position: &mut u64) -> Status,
     set_position: extern "win64" fn(this: &mut FileImpl, position: u64) -> Status,
-    get_info: usize,
-    set_info: usize,
+    get_info: unsafe extern "win64" fn(
+        this: &mut FileImpl,
+        information_type: &Guid,
+        buffer_size: &mut usize,
+        buffer: *mut u8,
+    ) -> Status,
+    set_info: unsafe extern "win64" fn(
+        this: &mut FileImpl,
+        information_type: &Guid,
+        buffer_size: usize,
+        buffer: *const c_void,
+    ) -> Status,
     flush: extern "win64" fn(this: &mut FileImpl) -> Status,
 }
 
-bitflags! {
-    /// Usage flags describing what is possible to do with the file.
-    pub struct FileMode: u64 {
-        /// The file can be read from.
-        const READ = 1;
-        /// The file can be written to.
-        const WRITE = 1 << 1;
-        /// The file will be created if not found.
-        const CREATE = 1 << 63;
-    }
+/// Usage flags describing what is possible to do with the file.
+///
+/// SAFETY: Using a repr(C) enum is safe here because this type is only sent to
+///         the UEFI implementation, and never received from it.
+///
+#[repr(u64)]
+pub enum FileMode {
+    /// The file can be read from
+    Read = 1,
+
+    /// The file can be read from and written to
+    ReadWrite = 2 | 1,
+
+    /// The file can be read, written, and will be created if it does not exist
+    CreateReadWrite = (1 << 63) | 2 | 1,
```

Comment:
Reading the UEFI spec reveals that only certain combinations of flags are allowed, so I think an enum makes more sense than exposing the raw flags.